panic: runtime error: index out of range [1] with length 1

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform Version

Latest 1.5 version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

terraform-provider-okta_v3.46.0 plugin

### Terraform Configuration Files


<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Debug Output

Error: Request cancelled
│ 
│ The plugin.(*GRPCProvider).ImportResourceState request was cancelled.
<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

Stack trace from the terraform-provider-okta_v3.46.0 plugin:
panic: runtime error: index out of range [1] with length 1
goroutine 125 [running]:
github.com/okta/terraform-provider-okta/okta.createCustomNestedResourceImporter.func1({0x103e240?, 0xc0001f0f60?}, 0xc000853080, {0xd?, 0xed4062?})
	github.com/okta/terraform-provider-okta/okta/utils.go:177 +0x238
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).ImportState(0xc0000a2180, {0x103e240, 0xc0001f0f60}, 0xc000093618, {0xc0004debb8, 0x14})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/provider.go:385 +0x1ac
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ImportResourceState(0xc000421920, {0x103e240?, 0xc0001f0060?}, 0xc00014aae0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:1088 +0xf4
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ImportResourceState(0xc0000ce0a0, {0x103e240?, 0xc000469560?}, 0xc0002b34a0)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:848 +0x1f6
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ImportResourceState_Handler({0xe8dbe0?, 0xc0000ce0a0}, {0x103e240, 0xc000469560}, 0xc0001413b0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:403 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00025e1e0, {0x10421d8, 0xc000487520}, 0xc0008cfd40, 0xc0000bbda0, 0x15e96b8, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1340 +0xd33
google.golang.org/grpc.(*Server).handleStream(0xc00025e1e0, {0x10421d8, 0xc000487520}, 0xc0008cfd40, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1713 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.51.0/server.go:963 +0x28a
Error: The terraform-provider-okta_v3.46.0 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

### Expected Behavior

<!--- What should have happened? --->

#### Can this be done in the Admin UI?

<!--- Yes / No / I don't know --->

#### Can this be done in the actual API call?

<!--- Yes / No / I don't know --->

### Actual Behavior

<!--- What actually happened? --->

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform plan`

### Important Factoids

This happened when I mistakenly had an import block without the authorisation server in the id

```
import {
 to = something
 id = 1234abc <-- forgot the auth server id prefix here
 }
```

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

- #0000

--- a/okta/resource_okta_auth_server_scope_test.go
+++ b/okta/resource_okta_auth_server_scope_test.go
@@ -3,6 +3,7 @@ package okta
 import (
 	"errors"
 	"fmt"
+	"regexp"
 	"strings"
 	"testing"
 
@@ -71,6 +72,27 @@ func TestAccResourceOktaAuthServerScope_crud(t *testing.T) {
 					}
 					return
 				},
+			},
+			{
+				// Addresses
+				// https://github.com/okta/terraform-provider-okta/issues/1759
+				// but benefits all resource imports that are compound input by
+				// concatenating input with slashes.
+				//
+				// Before fixing 1759 this step would cause the panic
+				// experienced in 1759. Now, it illustrates the provider will
+				// error if input was incorrect as just `auth_server_id` and not
+				// the expected `auth_server_id/id`.
+				ResourceName: resourceName,
+				ImportState:  true,
+				ImportStateIdFunc: func(s *terraform.State) (string, error) {
+					rs, ok := s.RootModule().Resources[resourceName]
+					if !ok {
+						return "", fmt.Errorf("failed to find %s", resourceName)
+					}
+					return rs.Primary.Attributes["auth_server_id"], nil
+				},
+				ExpectError: regexp.MustCompile(`expected 2 import fields "auth_server_id/id", got 1 fields "(\w*)"`),
 			},
 		},
 	})

--- a/okta/utils.go
+++ b/okta/utils.go
@@ -175,6 +175,9 @@ func createCustomNestedResourceImporter(fields []string, errMessage string) *sch
 	return &schema.ResourceImporter{
 		StateContext: func(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
 			parts := strings.Split(d.Id(), "/")
+			if len(parts) != len(fields) {
+				return nil, fmt.Errorf("expected %d import fields %q, got %d fields %q", len(fields), strings.Join(fields, "/"), len(parts), d.Id())
+			}
 			for i, field := range fields {
 				if field == "id" {
 					d.SetId(parts[i])

